orchestration/src/test/java/com/sap/ai/sdk/orchestration/SerializationTest.java

```diff
@@ -0,0 +1,139 @@
+                        .finishReason("stop")))
+            .usage(TokenUsage.create().completionTokens(7).promptTokens(19).totalTokens(26));
+
+    var orchestrationResult =
+        LLMModuleResult.create()
+            .id("chatcmpl-9lzPV4kLrXjFckOp2yY454wksWBoj")
+            ._object("chat.completion")
+            .created(1721224505)
+            .model("gpt-35-turbo-16k")
+            .choices(
+                List.of(
+                    LLMChoice.create()
+                        .index(0)
+                        .message(ChatMessage.create().role("assistant").content("General Kenobi!"))
+                        .finishReason("stop")))
+            .usage(TokenUsage.create().completionTokens(7).promptTokens(19).totalTokens(26));
+
+    var inputFilterResult =
+        GenericModuleResult.create()
+            .message("Input filter passed successfully.")
+            .data(
+                Map.of(
+                    "original_service_response",
+                    Map.of("Hate", 0, "SelfHarm", 0, "Sexual", 0, "Violence", 2),
+                    "checked_text",
+                    "Hello there!"));
+    var outputFilterResult =
+        GenericModuleResult.create()
+            .message("Output filter passed successfully.")
+            .data(
+                Map.of(
+                    "original_service_response",
+                    Map.of("Hate", 0, "SelfHarm", 0, "Sexual", 0, "Violence", 2),
+                    "checked_text",
+                    "General Kenobi!"));
+    var inputMaskingResult =
+        GenericModuleResult.create()
+            .message("Input to LLM is masked successfully.")
+            .data(
+                Map.of(
+                    "masked_template", List.of(Map.of("role", "user", "content", "Hello there!"))));
+    var outputMaskingResult =
+        List.of(
+            ModuleResultsOutputUnmaskingInner.create()
+                .index(0)
+                .message(ChatMessage.create().role("assistant").content("Hello there!"))
+                .finishReason("stop"));
+    var expected =
+        CompletionPostResponse.create()
+            .requestId("26ea36b5-c196-4806-a9a6-a686f0c6ad91")
+            .moduleResults(
+                ModuleResults.create()
+                    .templating(List.of(ChatMessage.create().role("user").content("Hello there!")))
+                    .llm(llmResult)
+                    .inputFiltering(inputFilterResult)
+                    .outputFiltering(outputFilterResult)
+                    .inputMasking(inputMaskingResult)
+                    .outputUnmasking(outputMaskingResult))
+            .orchestrationResult(orchestrationResult);
+
+    var actual =
+        OrchestrationClient.JACKSON.readValue(
+            getClass().getClassLoader().getResourceAsStream("serializedResponse.json"),
+            CompletionPostResponse.class);
+
+    assertThat(actual).isEqualTo(expected);
```

Comment:
This test needs to better reflect exceptions and stacktraces that the customer would encounter with a malformed response. Currently `JACKSON.readValue()` never throws with a malformed response.